---
title: "Heatmap Figure 6 Generation"
author: "Igor Todorovic, Jay Goddard"
date: "2024-03-26"
output: html_document
---

Generation of gene expression heatmap, derived from workflow in Neff and Doing, 2024 paper

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r packages, warning=FALSE, message=FALSE}
#install.packages("corrplot", dependencies=TRUE)
#BiocManager::install("ComplexHeatmap", dependencies=TRUE)

library(ComplexHeatmap)
library(corrplot)
library(stringi)
library(tidyverse)
```


Create heatmap of RNAseq values from samples recovered in Sputum and Synthetic Cystic Fibrosis Medium 2 (SCFM2)

*sputum and SCFM*

Read in data:

```{r data-import}
phoB_nonz<- read.csv("Sputum-v-SCFM2-samples_transcriptomics_phoB_regulon.csv")
```

Cleaning up the dataframe to extract the samples of interest

```{r data-tidying}
mnonz  <- as.matrix(phoB_nonz[c(2:5,8,10:13,15:18), c(1,3:37, 40:42)])
#mnonz  <- as.matrix(phoB_nonz[2:18, c(1, 3:44)])
colnames(mnonz) <- mnonz[1,]
rownames(mnonz) <- mnonz[,1]
# Remove unneeded character row and column
mnonz <- mnonz[-1, -1]

# Change class of matrix to numeric
class(mnonz) <- "numeric"
```
 
Next, we calculate z-scores to understand the data in terms of units of standard deviation:

```{r z-score_calculation}
Z <- t(scale(t(mnonz)))
```

We are using the function's default clustering for sample order.

If you would like to add an annotation so that the separation between Sputum and SCFM2 samples is more clear, you can do so

Create dataframe with the sample info:

```{r}
Sample_col <- data.frame(Sample = rep(c("Sputum", "SCFM2"), c(24, 14)))
row.names(Sample_col) <- colnames(Z)
```


```{r heatmap-generation}
#Z_sorted_ranked <- Z_sorted[,ranked_samples]
pheatmap(Z, breaks = c(-1,1), heatmap_legend_param = list(title="Z-score"), angle_col="45", fontsize_col=6, annotation_col = Sample_col)

```
